core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala

```diff
@@ -0,0 +1,21 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.models.PodSpec
+import org.apache.mesos.v1.{Protos => Mesos}
+
+/**
+  * Interface for the offer matcher implementations. Given a Mesos [[Mesos.Offer]] and a set of [[PodSpec]]s
+  * return a map of all matched PodSpecs with the list of matched Mesos resources.
+  */
+trait OfferMatcher {
```

Comment:
Is this premature generality? Can we just implement OfferMatcher and then extract it to be more generic in the future?

Reply:
I'd rather have a trait as a clear separation line. May be it's premature **now** but this part of the code tends to get entangles with the rest (especially when constraints are added)